File: codes/src-tgt/java-cpp/HumanEval_113_odd_count.cpp
### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static ArrayList<String> oddCount(ArrayList<String> lst) {
        ArrayList<String> res = new ArrayList<String>();
        for (String arr : lst) {
            int n = (int) Arrays.stream(arr.split("")).filter(d -> Integer.parseInt(d) % 2 == 1).count();
            res.add("the number of odd elements " + n + "n the str"+ n +"ng "+ n +" of the "+ n +"nput.");
        }
        return res;
    }
    

}


### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
std::vector<std::string> odd_count(std::vector<std::string> lst) {
    std::vector<std::string> res;
    for (auto arr : lst) {
        int n = std::count_if(arr.begin(), arr.end(), [](char c){return c%2==1;});
        res.push_back("the number of odd elements " + std::to_string(n) + "n the str"+ std::to_string(n) +"ng "+ std::to_string(n) +" of the "+ std::to_string(n) +"nput.");
    }
    return res;
}

File: dpo_train/generation_output/py-java/HumanEval_113_odd_count.java
### Python version

from typing import List

def odd_count(lst: List[str]) -> List[str]:
    res = []
    for arr in lst:
        n = sum(int(d)%2==1 for d in arr)
        res.append("the number of odd elements " + str(n) + "n the str"+ str(n) +"ng "+ str(n) +" of the "+ str(n) +"nput.")
    return res

### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static ArrayList<String> oddCount(ArrayList<String> lst) {
        ArrayList<String> res = new ArrayList<>();
        for (int i = 0; i < lst.size(); i++) {
            String str = lst.get(i);
            int n = str.length();
            if (n % 2 == 1) {
                res.add(str);
            }
        }
        return res;
    }
}

class Solution {
    public static void main(String[] args) {
        ArrayList<String> lst = new ArrayList<>();
        lst.add("the number of odd elements " + odd_count(lst));
        System.out.println(lst);
    }
}
